fixtures/tests/_fixtures/test_monkeypatch.py

```diff
@@ -310,16 +310,17 @@ def test_patch_boundmethod_with_boundmethod(self):
         with fixture:
             INST_C.foo()
         self.assertEqual(oldmethod, INST_C.foo)
+        sentinel = object()
+        self.assertEqual(sentinel, INST_C.__dict__.get('foo', sentinel))
 
     def test_patch_unboundmethod_with_staticmethod(self):
         oldmethod = C.foo
         fixture = MonkeyPatch(
             'fixtures.tests._fixtures.test_monkeypatch.C.foo',
             D.bar_static_args)
         with fixture:
-            c = C() 
-            tgtslf, arg = c.foo(1)
-            self.expectThat(tgtslf, Is(c))
+            tgtslf, arg = INST_C.foo(1)
```

Comment:
Not sure what you mean; the test is patching C.foo - the unbound method, and the original C.foo would return arg, not (self, arg), so we know it is indeed the patched in method (or a failed unpatch from an earlier test, I guess). Happy to change it to make a C() here again - this was how I discovered the failed cleanups of patching bound methods.